### PR TITLE
#397: Allow keywords to be used as external parameter names

### LINKED ISSUE #397: (Most) keywords allowed as external parameter names
[Apple has spoken](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/LexicalStructure.html#//apple_ref/doc/uid/TP40014097-CH30-ID413):

```
Keywords other than inout, var, and let can be used as external parameter names
in a function declaration or function call without being escaped with backticks.
```

I'm sure you are more than capable of fixing it but if you need a hint please give me a shout. Make sure that calling these is supported as well.

Test case:

```swift
extension SystemError {
    public static func description(for errorNumber: Int32) -> String {
        return String(cString: strerror(errorNumber))
    }
}

extension SystemError: CustomStringConvertible {
    public var description: String {
        return SystemError.description(for: errorNumber)
    }
}
```

### PR DESCRIPTION
**Pull request by a4sriniv**
Fixes #397.

### CODE REVIEW

File: src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4 [hunk 1]
```diff
@@ -856,12 +859,27 @@ NilLiteral: 'nil' ;
...
+keyword :
+ // Keywords used in declarations
+ 'associatedtype' | 'class' | 'deinit' | 'enum' | 'extension' | 'func' | 'import' | 'init' | 'internal'
+ | 'private' | 'protocol' | 'public' | 'static' | 'struct' | 'subscript' | 'typealias'
+ // Keywords used in statements
+ | 'break' | 'case' | 'continue' | 'default' | 'defer' | 'do' | 'else' | 'fallthrough' | 'for' | 'guard' | 'if' | 'in'
+ | 'repeat' | 'return' | 'switch' | 'where' | 'while'
+ // Keywords used in expressions and types
+ | 'as' | 'catch' | 'dynamicType' | 'is' | 'nil' | 'rethrows' | 'super' | 'self' | 'Self' | 'throw'
+ | 'throws' | 'try' | '#column' | '#file' | '#function' | '#line'
```
Comment (adityatrivedi) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: should `#file` etc be placed on line 875?

**Comment by adityatrivedi** [comment]
👍

File: src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4 [hunk 2]
```diff
@@ -856,12 +859,27 @@ NilLiteral: 'nil' ;
...
+ | '_'
+ // Keywords that begin with a number sign (#)
+ | '#available' | '#column' | '#else' | '#elseif' | '#endif' | '#file' | '#function' | '#if' | '#line' | '#selector'
+ ;
 
 contextSensitiveKeyword :
  'associativity' | 'convenience' | 'dynamic' | 'didSet' | 'final' | 'get' | 'infix' | 'indirect' |
  'lazy' | 'left' | 'mutating' | 'none' | 'nonmutating' | 'optional' | 'operator' | 'override' | 'postfix' | 'precedence' |
- 'prefix' | 'protocol' | 'required' | 'right' | 'set' | 'Type' | 'unowned' | 'weak' | 'willSet' |
+ 'prefix' | 'Protocol' | 'required' | 'right' | 'set' | 'Type' | 'unowned' | 'weak' | 'willSet' |
```
Comment (alykhank) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: Was this an incorrect change in the previous PR?

Comment (a4sriniv) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: yup

Comment (alykhank) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: Can we add a test case for it?

Comment (a4sriniv) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: done

**Comment by alykhank** [comment]
:+1:

**Pull request merged**